sympy/sets/fancysets.py

```diff
@@ -443,3 +445,309 @@ def _sup(self):
+                                   r*(cos(theta) + I*sin(theta))),
+                                   sets)
+        return obj
+
+    @property
+    def sets(self):
+        """
+        Return raw input sets to the self.
+
+        Examples
+        ========
+
+        >>> from sympy import Interval, ComplexPlane, Union
+        >>> a = Interval(2, 3)
+        >>> b = Interval(4, 5)
+        >>> c = Interval(1, 7)
+        >>> C1 = ComplexPlane(a*b)
+        >>> C1.sets
+        [2, 3] x [4, 5]
+        >>> C2 = ComplexPlane(Union(a*b, b*c))
+        >>> C2.sets
+        [2, 3] x [4, 5] U [4, 5] x [1, 7]
+
+        """
+        return self.args[1]
+
+    @property
+    def psets(self):
+        """
+        Return a tuple of sets (ProductSets) input of the self.
+
+        Examples
+        ========
+
+        >>> from sympy import Interval, ComplexPlane, Union
+        >>> a = Interval(2, 3)
+        >>> b = Interval(4, 5)
+        >>> c = Interval(1, 7)
+        >>> C1 = ComplexPlane(a*b)
+        >>> C1.psets
+        ([2, 3] x [4, 5],)
+        >>> C2 = ComplexPlane(Union(a*b, b*c))
+        >>> C2.psets
+        ([2, 3] x [4, 5], [4, 5] x [1, 7])
+
+        """
+        if self.args[1].is_ProductSet:
+            psets = ()
+            psets = psets + (self.args[1], )
+        else:
+            psets = self.args[1].args
+        return psets
+
+    @property
+    def a_interval(self):
+        """
+        Return the union of intervals of `x` when, self is in
+        rectangular form, or the union of intervals of `r` when
+        self is in polar form.
+
+        Examples
+        ========
+
+        >>> from sympy import Interval, ComplexPlane, Union
+        >>> a = Interval(2, 3)
+        >>> b = Interval(4, 5)
+        >>> c = Interval(1, 7)
+        >>> C1 = ComplexPlane(a*b)
+        >>> C1.a_interval
+        [2, 3]
+        >>> C2 = ComplexPlane(Union(a*b, b*c))
+        >>> C2.a_interval
+        [2, 3] U [4, 5]
+
+        """
+        a_interval = []
+        for element in self.psets:
+            a_interval.append(element.args[0])
+
+        a_interval = Union(*a_interval)
+        return a_interval
+
+    @property
+    def b_interval(self):
+        """
+        Return the union of intervals of `y` when, self is in
+        rectangular form, or the union of intervals of `theta`
+        when self is in polar form.
+
+        Examples
+        ========
+
+        >>> from sympy import Interval, ComplexPlane, Union
+        >>> a = Interval(2, 3)
+        >>> b = Interval(4, 5)
+        >>> c = Interval(1, 7)
+        >>> C1 = ComplexPlane(a*b)
+        >>> C1.b_interval
+        [4, 5]
+        >>> C2 = ComplexPlane(Union(a*b, b*c))
+        >>> C2.b_interval
+        [1, 7]
+
+        """
+        b_interval = []
+        for element in self.psets:
+            b_interval.append(element.args[1])
+
+        b_interval = Union(*b_interval)
+        return b_interval
+
+    @property
+    def polar(self):
```

Comment:
You can also assign `self.polar` to the `polar` argument in the `__new__` method, also the calculation is wrong because as both polar and cartesian form takes the input as Product sets.